src/networkextension.cs

```diff
@@ -148,6 +148,12 @@ enum NEFilterReportFrequency : long {
 		High,
 	}
 
+	[NoWatch, NoTV, NoiOS, Mac (10,15,5)]
+	[Native]
+	enum NEFilterDataAttribute : long {
```

Comment:
hmmm... where should that be used ?

Reply:
headers has some details

> 4-bytes NEFilterDataAttribute

so it should not be `[Native]` or `long` but Apple made it an `NSInteger` :(

Reply:
According to docs 

> @param readBytes The data that was read.  For non-UDP/TCP flows, since data may optionally include the IP header, readBytes includes a 4-bytes NEFilterDataAttribute field preceding the user data.  Handler must examine the NEFilterDataAttribute field and handle the data accordingly.

So it is a companion of these two https://github.com/xamarin/xamarin-macios/blob/8d3987b84a489e1ffd6fd825c1207b35f4a9fd32/src/networkextension.cs#L378-L382